File: cleanup.py
import logging
import os
import sqlite3
import time
from datetime import datetime
import pytz

logging.basicConfig(filename='cleanup.log', level=logging.DEBUG, format='%(asctime)s - %(levelname)s: %(message)s')

def delete_entries():
    conn = sqlite3.connect('db/database.db')
    cursor = conn.cursor()
    files_in_folder = os.listdir("uploads/")
    cursor.execute("SELECT name FROM files")
    files_in_database = [row[0] for row in cursor.fetchall()]
    for filename in files_in_database:
        if filename not in files_in_folder:
            cursor.execute("DELETE FROM files WHERE name =?", (filename,))
            logging.info(f"Deleted entry for file: {filename}")
            conn.commit()
    cursor.close()
    conn.close()

def delete_files():
    conn = sqlite3.connect('db/database.db')
    cursor = conn.cursor()
    cursor.execute('SELECT * FROM files')
    files = cursor.fetchall()
    current_time = int(datetime.now(pytz.timezone('UTC')).timestamp() * 1000)
    for file in files:
        name = file[0]
        upload_time = file[1]
        delete_time = file[2]
        if current_time > delete_time and delete_time != upload_time:
            file_path = os.path.join('uploads/', name)
            if os.path.exists(file_path):
                logging.info(
                    f"Deleted file: {name} \t Delete time in db(utc): {delete_time}")
                os.remove(file_path)
    cursor.close()
    conn.close()

if __name__ == '__main__':
    logging.debug('Started cleanup.py')
    while True:
        try:
            delete_files()
        except Exception as e:
            logging.error(e)
        try:
            delete_entries()
        except Exception as e:
            logging.error(e)
        time.sleep(60)

File: cleanup.php
<?php
require_once 'path.php';
date_default_timezone_set('UTC');
define('DB_FILE', DEFAULT_PATH . '/db/database.db');
ini_set('error_log', 'cleanup.log');


function delete_entries()
{
    $conn = new PDO('sqlite:' . DB_FILE);
    $files_in_folder = array();
    $files_in_database = array();
    if (is_dir("uploads/")) {
        $directory = new DirectoryIterator("uploads/");
        foreach ($directory as $file) {
            if ($file->isFile()) {
                $files_in_folder[] = $file->getFilename();
            }
        }
    }
    try {
        $pdo = new PDO("sqlite:" . DB_FILE);
        $pdo->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
        $query = "SELECT name FROM files";
        $stmt = $pdo->query($query);
        while ($row = $stmt->fetch(PDO::FETCH_ASSOC)) {
            $files_in_database[] = $row['name'];
        }
    } catch (PDOException $e) {
        echo $e;
    }
    foreach ($files_in_database as $filename) {
        if (!in_array($filename, $files_in_folder)) {
            $stmt = $conn->prepare("DELETE FROM files WHERE name = :filename");
            $stmt->bindParam(':filename', $filename);
            $stmt->execute();
            error_log("Deleted entry for file:" . $filename . PHP_EOL, 3, 'cleanup.log');
        }
    }
    $conn = null;
}

function delete_files()
{
    try {
        $conn = new PDO('sqlite:' . DB_FILE);;
        $stmt = $conn->query("SELECT * FROM files");
        $current_time = time() * 1000;
        while ($row = $stmt->fetch(PDO::FETCH_ASSOC)) {
            $name = $row['name'];
            $upload_time = $row['uploadTime'];
            $delete_time = $row['deleteTime'];
            if ($current_time > $delete_time && $delete_time != $upload_time) {
                $file_path = 'uploads/' . $name;
                if (file_exists($file_path)) {
                    error_log("Deleted file: " . $name . "\t Delete time in db(utc): " . $delete_time . PHP_EOL, 3, 'cleanup.log');
                    unlink($file_path);
                }
            }
        }
    } catch (PDOException $e) {
        echo $e;
    }
    $conn = null;
}

try {
    delete_files();
} catch (Exception $e) {
    error_log($e, 3, 'cleanup.log');
}

try {
    delete_entries();
} catch (Exception $e) {
    error_log($e, 3, 'cleanup.log');
}
